### PR TITLE
Return AdjFloat of an AdjFloat summation

### PR DESCRIPTION
**Pull request by Ig-dolci**

### CODE REVIEW

File: pyadjoint/adjfloat.py [hunk 1]
```diff
@@ -342,7 +342,10 @@ def __init__(self, *args):
             self.add_dependency(dep)
 
     def recompute_component(self, inputs, block_variable, idx, prepared):
-        return self.operator(*(term.saved_output for term in self.terms))
+        output = self.operator(*(term.saved_output for term in self.terms))
+        if isinstance(output, float):
+            return AdjFloat(output)
+        return output
```
Comment (Ig-dolci) [inline pyadjoint/adjfloat.py:348]:
```suggestion
        return self.terms[0].saved_output._ad_convert_type(output)
```

File: pyadjoint/adjfloat.py [hunk 2]
```diff
@@ -342,7 +342,8 @@ def __init__(self, *args):
             self.add_dependency(dep)
 
     def recompute_component(self, inputs, block_variable, idx, prepared):
-        return self.operator(*(term.saved_output for term in self.terms))
+        output = self.operator(*(term.saved_output for term in self.terms))
+        return self.terms[0].saved_output._ad_convert_type(output)
```
Comment (Ig-dolci) [inline pyadjoint/adjfloat.py:346]: The type should be the same as Block output.

Comment (Ig-dolci) [reply pyadjoint/adjfloat.py]: done

File: pyadjoint/reduced_functional.py [hunk 3]
```diff
@@ -196,6 +197,23 @@ def __call__(self, values):
         if len(values) != len(self.controls):
             raise ValueError("values should be a list of same length as controls.")
 
+        for i, value in enumerate(values):
+            if isinstance(value, (int, float)):
+                value = AdjFloat(value)
```
[the remark this answers is not in the document]
Comment (Ig-dolci) [reply pyadjoint/reduced_functional.py]: done

**Review by connorjward — APPROVED** [review]

**Pull request merged**